Sources/VergeCore/GetterBuilderMethodChain.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  GetterBuilderMethodChain.swift
+//  VergeCore
+//
+//  Created by muukii on 2020/03/27.
+//  Copyright © 2020 muukii. All rights reserved.
+//
+
+import Foundation
+
+public enum GetterBuilderTrait {
+  public enum Rx {}
+  public enum Combine {}
+}
+
+// MARK: - Method Chain
+
+public struct GetterBuilderMethodChain<Trait, Container: ValueContainerType> {
+  
+  public typealias Input = Container.Value
+  
+  private let target: Container
+  var onPreFilterWillReceive: ((Input) -> Void) = { _ in }
+  
+  public init(target: Container) {
+    self.target = target
+  }
+  
+  public func `do`(onReceive: @escaping (Input) -> Void) -> Self {
+    var _self = self
+    let pre = onPreFilterWillReceive
+    _self.onPreFilterWillReceive = { value in
+      pre(value)
+      onReceive(value)
+    }
+    return _self
+  }
```

Comment:
Most of the use cases are debugging.
In order to know the number of events that receiving events from storage.